src/core/container_async.rs

```diff
@@ -0,0 +1,122 @@
+use crate::core::{DockerAsync, ImageAsync, LogsAsync};
+use futures::executor::block_on;
+use std::env::var;
+
+/// Represents a running docker container using async trait.
+///
+/// Containers have a [`custom destructor`][drop_impl] that removes them as soon as they
+/// go out of scope. However, async drop is not available in rust yet. This implementation
+/// is using block_on. therefore required tokio::test(threaded_scheduler) in your test
+/// to use drop effectively. Otherwise your test might stall:
+///
+/// ```rust
+/// use testcontainers::*;
+/// #[tokio::test(threaded_scheduler)]
+/// async fn a_test() {
+///     let docker = clients::Http::default();
+///
+///     {
+///         let container = docker.run(MyImage::default());
+///
+///         // Docker container is stopped/removed at the end of this scope.
+///     }
+/// }
+///
+/// ```
+///
+/// [drop_impl]: struct.ContainerAsync.html#impl-Drop
+#[derive(Debug, Clone)]
+pub struct ContainerAsync<'d, D, I>
+where
+    D: DockerAsync,
+    I: ImageAsync,
+{
+    id: String,
+    docker_client: &'d D,
+    image: I,
+}
+
+impl<'d, D, I> ContainerAsync<'d, D, I>
+where
+    D: DockerAsync,
+    I: ImageAsync,
+{
+    /// Constructs a new container given an id, a docker client and the image.
+    /// ContainerAsync::new().await
+    pub async fn new(id: String, docker_client: &'d D, image: I) -> ContainerAsync<'d, D, I> {
+        let container = ContainerAsync {
+            id,
+            docker_client,
+            image,
+        };
+
+        container.block_until_ready().await;
+
+        container
+    }
+
+    /// Returns the id of this container.
+    pub fn id(&self) -> &str {
+        &self.id
+    }
+
+    /// Gives access to the log streams of this container.
+    pub async fn logs<'a>(&'a self) -> LogsAsync<'a> {
+        let id_clone: String = String::from(&self.id);
+        self.docker_client
+            .logs(Box::leak(id_clone.into_boxed_str()))
```

Comment:
This `Box::leak` is a bit concerning. I think we have a memory leak here if I am reading the [docs](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak) correctly.

Feel free to change the signature of `.logs` if you don't want to mess around with manual dropping here.